### PR TITLE
docker setup, creation of django project

### PR DESCRIPTION
**Pull request by maria-j-k**
I'm not sure about creating and activating virtual environment. Without I was getting the following warning:
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

I was creating files following Docker documentation (https://docs.docker.com/samples/django/), where there was no mention about virtual environment

### CODE REVIEW

File: trello/settings.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+# Build paths inside the project like this: BASE_DIR / 'subdir'.
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/3.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'django-insecure-r(i@i*pd)iw2cni*$v4yzm&1k#4n_vb32it)znzr5o_k-i+_@s'
```
Comment (nitz14) [inline trello/settings.py]: SECURITY WARNING: keep the secret key used in production secret!

File: trello/settings.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
```
Comment (nitz14) [inline trello/settings.py]: This variable should be taken from env variable

File: trello/settings.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+WSGI_APPLICATION = 'trello.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/3.2/ref/settings/#databases
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.sqlite3',
```
Comment (nitz14) [inline trello/settings.py]: we want to use postgresql not sqlite3

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,12 @@
+FROM python:3
+
+ENV PYTHONUNBUFFERED=1
+
+ENV VIRTUAL_ENV=/venv
+RUN python3 -m venv $VIRTUAL_ENV
```
Comment (nitz14) [inline Dockerfile]: You don't need to create venv inside in container.

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+version: "3.9"
+
+services:
+    web:
```
Comment (nitz14) [inline docker-compose.yml:17]: I think we miss few things here for example database, possible to pass env variable to project.

**Review by nitz14 — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,12 @@
+FROM python:3
+
+ENV PYTHONUNBUFFERED=1
+
+ENV VIRTUAL_ENV=/venv
```
Comment (robertmusil) [inline Dockerfile]: If you want to get rid of that pip warning it's better to just switch users in container
```
RUN useradd -u 100 python
USER python
```

File: trello/settings.py (hunk 3 above)
Comment (robertmusil) [inline trello/settings.py]: Please use `postgres` for a database, I think it's our fault that we forgot to mention it.

File: trello/settings.py (hunk 1 above)
Comment (robertmusil) [inline trello/settings.py]: This should come form env varriables and not be commited

File: requirements.txt [hunk 7]
```diff
@@ -0,0 +1,2 @@
+Django>=3.0,<4.0
+psycopg2-binary>=2.0
```
Comment (robertmusil) [inline requirements.txt:2]: Why do you have `psycopg2` if you're using sqlite? (though you should be using postgres)

**Review by robertmusil — CHANGES_REQUESTED** [review]

File: trello/settings.py (hunk 1 above)
Comment (maria-j-k) [reply trello/settings.py]: done

File: trello/settings.py (hunk 3 above)
Comment (maria-j-k) [reply trello/settings.py]: done

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,30 @@
+version: "3.9"
+
+services:
+    db: 
+        image: postgres
+        volumes:
+            - ./data/db:/ver/lib/postgresql/data
```
Comment (nitz14) [inline docker-compose.yml]: try without this volumes

File: trello/settings.py [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+For the full list of settings and their values, see
+https://docs.djangoproject.com/en/3.2/ref/settings/
+"""
+
+from pathlib import Path
+import environ
+import os
+
+
+env = environ.Env(
```
Comment (nitz14) [inline trello/settings.py:18]: So what I should provide by DEBUG env value to turn off DEBUG mode in django ?

Comment (maria-j-k) [reply trello/settings.py:18]: To have DEBUG enabled, you must set it to one of boolean true strings ('yes', 'y', 'on', 'true, '1'). So any other value (or lack of the value) will result in default setting of DEBUG=False

File: Dockerfile [hunk 10]
```diff
@@ -0,0 +1,17 @@
+FROM python:3
+RUN apt-get update \
+    && apt-get install -y postgresql-client \
```
Comment (robertmusil) [inline Dockerfile:3]: Why would you need postgresql-client?

File: .env-template [hunk 11]
```diff
@@ -0,0 +1,6 @@
+DEBUG=on
+SECRET_KEY=your-secret-key
+DATABASE_URL=psql://user:un-githubbedpassword@127.0.0.1:8458/database
```
Comment (robertmusil) [inline .env-template:3]: Ideally I'd want to copy paste this template to create dev env out of the box, here you have variables that are connected, but have different values from each other (db username, password)

File: trello/settings.py [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+            DEBUG=(bool, False)
+        )
+# Build paths inside the project like this: BASE_DIR / 'subdir'.
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+environ.Env.read_env(os.path.join(BASE_DIR, '.env'))
```
Comment (robertmusil) [inline trello/settings.py:24]: Why do you need to read .env file, its being set already by docker-compose

**Review by robertmusil — APPROVED** [review]

**Review by nitz14 — APPROVED** [review]
LF

File: .env-template (hunk 11 above)
Comment (maria-j-k) [reply .env-template:3]: I'll corect it in next commit.

File: Dockerfile (hunk 10 above)
Comment (maria-j-k) [reply Dockerfile:3]: Python needs it to communicate with Postgresql

File: trello/settings.py (hunk 12 above)
Comment (maria-j-k) [reply trello/settings.py:24]: 1. I was following the instructions in django-environ2 documentation
2. I thought I have to tell django settings where environment variables are stored
3. Is the fact of having in working directory .env file sufficient to effectively export it? So I can use in settings.py os.environ.get(variable_name)?

**Pull request merged**